sql/src/main/java/io/crate/planner/TableStatsService.java

```diff
@@ -61,25 +60,27 @@
 
     private final ClusterService clusterService;
     private final Provider<SQLOperations> sqlOperationsProvider;
+    private final ThreadPool threadPool;
+    private final NodeSettingListener nodeSettingListener = new NodeSettingListener();
     private volatile ObjectLongMap<TableIdent> tableStats = null;
-
-    @BindingAnnotation
-    @Target({ElementType.FIELD, ElementType.PARAMETER, ElementType.METHOD})
-    @Retention(RetentionPolicy.RUNTIME)
-    public @interface StatsUpdateInterval {
-    }
-
+    private ThreadPool.Cancellable refreshScheduledTask = null;
+    private TimeValue initialRefreshInterval;
+    private TimeValue lastRefreshInterval;
 
     @Inject
     public TableStatsService(Settings settings,
                              ThreadPool threadPool,
                              ClusterService clusterService,
-                             @StatsUpdateInterval TimeValue updateInterval,
+                             NodeSettingsService nodeSettingsService,
                              Provider<SQLOperations> sqlOperationsProvider) {
         super(settings);
+        this.threadPool = threadPool;
         this.clusterService = clusterService;
         this.sqlOperationsProvider = sqlOperationsProvider;
-        threadPool.scheduleWithFixedDelay(this, updateInterval, ThreadPool.Names.REFRESH);
+        initialRefreshInterval = extractRefreshInterval(settings);
```

Comment:
It's needed to be able to go back to the initial setting when a reset global is issued